iot-hub/Samples/device/PnpDeviceSamples/TemperatureController/TemperatureControllerSample.cs

```diff
@@ -68,8 +68,8 @@ public class TemperatureControllerSample
 
         public TemperatureControllerSample(DeviceClient deviceClient, ILogger logger)
         {
-            _deviceClient = deviceClient ?? throw new ArgumentNullException($"{nameof(deviceClient)} cannot be null.");
-            _logger = logger ?? LoggerFactory.Create(builer => builer.AddConsole()).CreateLogger<TemperatureControllerSample>();
+            _deviceClient = deviceClient ?? throw new ArgumentNullException(nameof(deviceClient));
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
```

Comment:
Q - any reason why we're opting to not create a default logger, in case one doesn't already exist?

Reply:
1. It is created in Program.cs. There's no chance one won't exist in this sample.
2. This was a dead code path, as a result.
3. If this code path were to be exercised, it included an undisposed object resulting in a memory leak.
4. Why not treat this parameter the same as the others? The sample needs these things to be successfully run and be useful to an observer.